Improved wake behaviour

SchemingWeasels opened the pull request:
## Issues
 - Resolves #82 

## Description
Reduced screen wakes to only relevant events

## Types of changes
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [X] Requested changes are in a branch
- [X] Compiled and tested requested changes on target hardware (PC, device)
- [X] Updated the documentation, if necessary
- [ ] Updated the LICENSES file, if necessary
- [X] Reviewed the [guidelines for contributing](../CONTRIBUTING.md) to this repository


## Further comments

Not sure if changing the way the command decode is ideal. I did it this way because I need access to the incoming command.
The other option would be to put in the wake-checking code into ProcessPackage, which seems like a worse idea to me. Thoughts?

SchemingWeasels committed:
Improved wake behaviour

Reduced screen wakes to only relevant events

SchemingWeasels committed:
Updated function briefs

t3knomanzer commented on Embedded/MaxMix/MaxMix.ino:
```diff
@@ -513,3 +523,23 @@ int8_t FindItem(uint32_t id)
+// Checks if command package target is the active application.
+// Returns true if ids match.
+//---------------------------------------------------------
+bool IsActive()
```
I would probably rename this to something a bit less generic such as IsCommandItemActive or something like that.

t3knomanzer commented on Embedded/MaxMix/MaxMix.ino:
```diff
@@ -128,11 +128,23 @@ void loop()
+      {
+        UpdateActivityTime();
+        isDirty = true;
+      }
```
I agree with your observation. I would probably rename UpdateActivityTime to UpdateActivity and move all this logic inside.

t3knomanzer reviewed (commented):
@XScorpion2 This was your request, what do you think?

XScorpion2 commented:
So I haven't spent a whole lot of time looking at the firmware code layout, but over all the approach makes sense so I'm good with it.

t3knomanzer commented:
Ok. I'm going to wait to merge this until after we release 0.10.5. I'm currently testing it and fixing some bugs introduced before releasing it.

SchemingWeasels committed:
Restructured ProcessPackage

Display update logic is done in ProcessPackage since checks are already being done on relevantvariables.

ProcessPackage returns true if a screen update is required.

Improved function naming and documentation.

SchemingWeasels commented:
I have completely restructured the way the checks are performed. ProcessPackage returns true if an update is required, just like the other process functions. Having said what I said in the original post, I think this is a cleaner approach, and it kinda makes sense to make ProcessPackage flag if a screen update is required. 

What do you think?

I also renamed the functions to be more descriptive as suggested and did some documentation for the functions I changed.

SchemingWeasels commented:
> and fixing some bugs introduced before releasing it.

Is it on the firmware side?

t3knomanzer commented:
No, it’s on the application side. We introduced process grouping which involved a pretty big rewrite of the audio service and that’s introduced some issues. 
It’s almost done now.

t3knomanzer commented:
Hey, can you rebase on master so I can test it with the latest changes?
